portal-impl/src/com/liferay/portal/service/impl/AddressServiceImpl.java

```diff
@@ -138,4 +138,16 @@ public Address updateAddress(
 			regionId, countryId, listTypeId, mailing, primary, phoneNumber);
 	}
 
+	public Address updateExternalReferenceCode(
+			Address address, String externalReferenceCode)
+		throws PortalException {
+
+		CommonPermissionUtil.check(
+			getPermissionChecker(), address.getClassNameId(),
+			address.getClassPK(), ActionKeys.UPDATE);
+
```

Comment:
I think this is not enough compared with the updateAddress method where we check another permission if it's an address related to an account